src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -745,25 +744,37 @@ private Transition getTransitionByName(Iterable<Transition> transitions, String
         return null;
     }
 
-    private void loadCustomFields() {
+    private void loadCustomFields(String jiraProject, String issueType, boolean force) {
         log.debug("Loading all custom fields");
+        Map<String, String> fields = new HashMap<>();
+        if(!force && this.customFields.containsKey(jiraProject.concat(issueType))){
+            return;
+        }
         GetCreateIssueMetadataOptions options;
         options = new GetCreateIssueMetadataOptionsBuilder()
                 .withExpandedIssueTypesFields()
+                .withProjectKeys(jiraProject)
+                .withIssueTypeNames(issueType)
                 .build();
         Iterable<CimProject> metadata = this.issueClient.getCreateIssueMetadata(options).claim();
         CimProject cim = metadata.iterator().next();
         cim.getIssueTypes().forEach( issueTypes -> {
             issueTypes.getFields().forEach((id, value) -> {
                 String name = value.getName();
-                this.customFields.put(name, id);
+                fields.put(name, id);
             });
         });
+        this.customFields.put(jiraProject.concat(issueType), fields);
```

Comment:
Not really an issue, but I would recommend doing the concatenation only once at the beginning of the method
jiraProject.concat(issueType)

Reply:
Am I missing something?  This step is only executed once - it is not within the stream or loop.